api/spec/services/company_holiday_calculator_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CompanyHolidayCalculator, type: :service do
+  describe '#calculate_company_holidays_during_service' do
+    let(:beginning) { Date.parse('2017-12-01') }
+    let(:ending) { Date.parse('2018-01-31') }
+    let(:company_holiday_calculator) { CompanyHolidayCalculator.new(beginning, ending) }
+
+    before do
+      create :holiday, beginning: '2018-01-01', ending: '2018-01-07'
+    end
+
+    context 'when there are no public holidays' do
+      context 'when there is one company holiday' do
+        it { expect(company_holiday_calculator.calculate_company_holiday_days_during_service).to eq 5 }
```

Comment:
Maybe move this to a subject

```ruby
subject { company_holiday_calculator.calculate_company_holiday_days_during_service }
```

and then 

```suggestion
        it { is_expected.to eq 5 }
```